pkg/apiserver/rest/dualwriter.go

```diff
@@ -181,6 +186,8 @@ func SetDualWritingMode(
 		}
 	}
 
+	dualWriter := NewDualWriter(currentMode, legacy, storage, reg, group, resource, namespacer)
```

Comment:
Hmm I think this might prevent the correct current mode from getting set.

We rely on the value of "currentMode" in order to pick the correct dual writer to initialise. The current mode only gets set in the "if" blocks below though. For example, consider:

```
	if (desiredMode == Mode2) && (currentMode == Mode1)
```

The dual writer we want to initialise is a mode 2 dual writer. However, on this line, current mode is still mode 1 if the kvstore doesn't contain a current mode entry. Therefore we will end up initialising a mode 1 dual writer not mode 2 as desired.